### PR TITLE
[layer] Bug fixes for layers

### PR DESCRIPTION
**Pull request by kparichay**
This patch adds bug fixes for layers:
- bn layer support in-place is corrected.
- flatten layer support in-place is corrected.
- pooling layer padding related corrections.

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

### CODE REVIEW

File: nntrainer/layers/pooling2d_layer.cpp [hunk 1]
```diff
@@ -92,8 +93,8 @@ void Pooling2DLayer::finalize(InitLayerContext &context) {
 
   unsigned int IM = std::numeric_limits<int>::max();
 
-  if (eff_in_height - pb - pool_size[0] > IM ||
-      eff_in_width - pr - pool_size[1] > IM) {
+  if (eff_in_height - pt - pool_size[0] > IM ||
+      eff_in_width - pl - pool_size[1] > IM) {
```
Comment (zhoonit) [inline nntrainer/layers/pooling2d_layer.cpp:97]: Nice catch!

**Review by zhoonit — APPROVED** [review]
LGTM!

**Review by lhs8928 — APPROVED** [review]
LGTM

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Pull request merged**